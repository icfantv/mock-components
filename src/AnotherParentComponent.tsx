import * as React from 'react'
import { BindAll } from 'lodash-decorators'
import { ChildComponent } from 'ChildComponent'

@BindAll()
export class AnotherParentComponent extends React.Component {
  public render(): React.ReactElement<AnotherParentComponent> {
    return (
      <div>
        <div>Another Parent Component</div>
        <div>
          <ChildComponent />
        </div>
      </div>
    )
  }
}
