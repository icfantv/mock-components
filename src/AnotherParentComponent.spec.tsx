import * as React from 'react'
import { create } from 'react-test-renderer'

import { AnotherParentComponent } from 'AnotherParentComponent'

describe('test the <AnotherParentComponent/> component', () => {
  test('it should deep render another parent component', () => {
    const tree = create(<AnotherParentComponent />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
