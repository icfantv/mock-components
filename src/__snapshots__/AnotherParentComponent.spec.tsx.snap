// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the <AnotherParentComponent/> component it should deep render another parent component 1`] = `
<div>
  <div>
    Another Parent Component
  </div>
  <div>
    <div>
      Child Component
    </div>
  </div>
</div>
`;
